chain/bitcoind_client.go

```diff
@@ -52,14 +52,12 @@ type BitcoindClient struct {
 	bestBlockMtx sync.RWMutex
 	bestBlock    waddrmgr.BlockStamp
 
-	// rescanUpdate is a channel will be sent items that we should match
-	// transactions against while processing a chain rescan to determine if
-	// they are relevant to the client.
-	rescanUpdate chan interface{}
-
 	// watchedAddresses, watchedOutPoints, and watchedTxs are the set of
 	// items we should match transactions against while processing a chain
 	// rescan to determine if they are relevant to the client.
+	//
+	// TODO(yy): make sure to GC these maps once the spending tx has
```

Comment:
👍 

Let's just roll this out into an issue now?

Reply:
I actually don't think we can ever remote entries in this map. We essentially keep an in memory map of every addr/utxo we might ever want to notify off of. If someone ends to an addr we used last 1 year ago, we still want to generate a notification. 

So one way forward here would be to spin this out into an interface, so we have more flexibility here w.r.t caching and/or implementation.

Reply:
> This got me to think about the restart behavior, I think watchedAddresses should be updated during restart?

During restart, we'll fetch all the relevant addr/UTXOs using `activeData`: https://github.com/btcsuite/btcwallet/blob/b2cb575cc6ccc0c41cf24ea834df580414b7acac/wallet/wallet.go#L557-L572

Reply:
Q: An active address is bascially every address in our addr db ?

Reply:
> Q: An active address is bascially every address in our addr db ?

I think so.